key/stringify.go

```diff
@@ -46,19 +47,37 @@ func Stringify(key interface{}) (string, error) {
 		str = key
 	case map[string]interface{}:
 		keys := SortedKeys(key)
-		for _, k := range keys {
+		var err error
+		for i, k := range keys {
 			v := key[k]
-			if len(str) > 0 {
-				str += "_"
-			}
-			s, err := Stringify(v)
+			keys[i], err = Stringify(v)
```

Comment:
Why only the values here, incidientally?  I note for the `map[Key]interface{}` case, we are including the keys.